polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -91,6 +91,11 @@ bool isLLVMStructABI(const RecordDecl *RD, llvm::StructType *ST) {
       if (m->isVirtualAsWritten() || m->isPure())
         return true;
     }
+    for (const auto &Base : CXRD->bases()) {
+      const CXXRecordDecl *BaseDecl = Base.getType()->getAsCXXRecordDecl();
```

Comment:
We probably can skip using this temp variable and directly use inside the if statement?

Reply:
Do you feel strongly about this? I prefer to have a temp variable because
(1) it is consistent with clang's implementation
(2) it makes it clearer that the first part is getting the CXXRecordDecl, and the logic is to simply check if it is empty.

Reply:
The function name `getAsCXXRecordDecl` makes it pretty clear that the return type is `CXXRecordDecl`.  What do you think?  I do not think being consistent with clang for this small piece of code is much important.

Reply:
ok